tests/extras/datasets/spark/test_spark_dataset.py

```diff
@@ -159,6 +176,66 @@ def test_load_options_csv(self, tmp_path, sample_pandas_df):
         spark_df = spark_data_set.load()
         assert spark_df.filter(col("Name") == "Alex").count() == 1
 
+    def test_load_options_schema_ddl_string(
+        self, tmp_path, sample_pandas_df, sample_spark_df_schema
+    ):
+        filepath = (tmp_path / "data").as_posix()
+        local_csv_data_set = CSVDataSet(filepath=filepath)
+        local_csv_data_set.save(sample_pandas_df)
+        spark_data_set = SparkDataSet(
+            filepath=filepath,
+            file_format="csv",
+            load_args={"header": True, "schema": "name STRING, age INT, height FLOAT"},
+        )
+        spark_df = spark_data_set.load()
+        assert spark_df.schema == sample_spark_df_schema
+
+    def test_load_options_schema_obj(
+        self, tmp_path, sample_pandas_df, sample_spark_df_schema
+    ):
+        filepath = (tmp_path / "data").as_posix()
+        local_csv_data_set = CSVDataSet(filepath=filepath)
+        local_csv_data_set.save(sample_pandas_df)
+
+        spark_data_set = SparkDataSet(
+            filepath=filepath,
+            file_format="csv",
+            load_args={"header": True, "schema": sample_spark_df_schema},
+        )
+
+        spark_df = spark_data_set.load()
+        assert spark_df.schema == sample_spark_df_schema
+
+    def test_load_options_schema_path(
+        self, tmp_path, sample_pandas_df, sample_spark_df_schema
+    ):
+        filepath = (tmp_path / "data").as_posix()
+        schemapath = (tmp_path / "schema.json").as_posix()
+        local_csv_data_set = CSVDataSet(filepath=filepath)
+        local_csv_data_set.save(sample_pandas_df)
+        Path(schemapath).write_text(sample_spark_df_schema.json(), encoding="utf-8")
+
+        spark_data_set = SparkDataSet(
+            filepath=filepath,
+            file_format="csv",
+            load_args={"header": True, "schema": {"filepath": schemapath}},
+        )
+
+        spark_df = spark_data_set.load()
+        assert spark_df.schema == sample_spark_df_schema
+
+    def test_load_options_schema_path_err(self, tmp_path):
+        filepath = (tmp_path / "data").as_posix()
+        schemapath = (tmp_path / "schema.json").as_posix()
+        Path(schemapath).write_text("dummy", encoding="utf-8")
+
+        with pytest.raises(DataSetError):
```

Comment:
We check the type of exception raised _and_ the message it was raised with. Can you add a `match=` here as well for that please?

Reply:
Done! 🚀